src/fides/api/service/connectors/mongodb_connector.py

```diff
@@ -44,7 +44,7 @@ def create_client(self) -> MongoClient:
         """Returns a client for a MongoDB instance"""
         uri = (self.configuration.secrets or {}).get("url") or self.build_uri()
         try:
-            return MongoClient(uri, serverSelectionTimeoutMS=5000)
+            return MongoClient(uri, serverSelectionTimeoutMS=5000, uuidRepresentation='pythonLegacy')
```

Comment:
In pymongo version 4, [default UUID representation changes](https://pymongo.readthedocs.io/en/stable/migrate-to-pymongo4.html#encoding-a-uuid-raises-an-error-by-default) from 'pythonLegacy' to 'unspecified', which raises an error when encoding UUIDs.

I've added the 'pythonLegacy' string to keep the existing behavior, I'd love an extra pair of eyes on the [relevant mongo docs ](https://pymongo.readthedocs.io/en/stable/examples/uuid.html#supported-uuid-representations) so that we can determine if this is the best solution. TBH I didn't understand all of the doc. 

Another things is that according to the docs, we should be able to pass in a constant to `uuidRepresentation`, like
```py
from bson.binary import UuidRepresentation
client = MongoClient(uuidRepresentation=UuidRepresentation.PYTHON_LEGACY)
```
however when I tried it, it turns out the field expects a string but `UuidRepresentation.PYTHON_LEGACY` is a number so I'm not sure what I should do there 🤔

Reply:
Thanks for flagging! What's tricky here is that we don't know which uuid subtypes our customers are using.  Different customers could use different uuid representations. We could make this configurable, but since we don't officially support uuid types yet, I don't think this is worth pursuing at this time unless it is a customer request.  Instead, I think we should pick one for now, as you've done here, but I'd lean towards "standard" instead, as I think this would be more common.  I'd also like to see test coverage against a UUID type since we don't have this data type in our sample mongo data.  

So next steps, something like:

1) Update to `UuidRepresentation.standard` here
2) Add a `customer_uuid` field or similar as a standard UUID type to `customer_details` records in `mongo_sample.js`, to automatically add this data on startup in our example database, for example:
```js
db.customer_details.insert([
    {
        "customer_id": 1,
        "customer_uuid": new UUID('3b241101-e2bb-4255-8caf-4136c566a962'),
        "gender": "male"
        ...
   }
])
```
3) Update `data/dataset/mongo_example_test_dataset.yml` to add this `customer_uuid` field to the customer_details table, and also give it a "user." data type so Fides will know to target it in DSR's:
```yaml
  - name: customer_details
    fields:
    ...
    - name: customer_uuid
      data_categories:
        - user.unique_id
```
4) Update one of our end-to-end tests for mongo access requests to verify this customer uuid is serialized properly.   I think `tests/ops/integration_tests/test_mongo_task.py > test_object_querying_mongo` would be a good one because it uses real datasets.  `target_categories` would need to be updated to also look for `user.unique_id`, and then you could assert the value of `customer_uuid` is what you expect.
5) Also complete end-to-end DSR testing similar to your "Steps to confirm" above, running a DSR on a table with a UUID